src/model/wallet.ts

```diff
@@ -491,9 +531,19 @@ export const oldLoadSeedPhrase = async (): Promise<null | EthereumWalletSeed> =>
 export const loadAddress = (): Promise<null | EthereumAddress> =>
   keychain.loadString(addressKey) as Promise<string | null>;
 
-const loadPrivateKey = async (
+export const loadPrivateKey = async (
   address?: EthereumAddress | undefined
 ): Promise<null | EthereumPrivateKey | -1 | -2> => {
+  const addressToUse = address || (await loadAddress());
+  if (!addressToUse) {
+    return null;
+  }
+
+  // checks if the address is a hardware wallet for proper handling
+  const { wallets } = store.getState().wallets;
+  const selectedWallet = findWalletWithAccount(wallets!, addressToUse);
+  const isHardwareWallet = selectedWallet?.type === walletTypes.bluetooth;
```

Comment:
`bluetooth` is a not a great walletType name imo cause you're assuming ledger nano x is the only BT device you'll ever integrate

Reply:
using type for handling and walletType for specifics so I think this works 
`{type: bluetooth, walletType: ledger}`